src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs

```diff
@@ -19,15 +19,13 @@
 namespace Microsoft.Win32.SafeHandles
 {
     [SecurityCritical]
-    internal sealed class SafeTokenHandle : SafeHandle
+    public sealed class SafeTokenHandle : SafeHandle
     {
         private const int DefaultInvalidHandleValue = 0;
 
-        internal static readonly SafeTokenHandle InvalidHandle = new SafeTokenHandle(new IntPtr(DefaultInvalidHandleValue));
+        private SafeTokenHandle() : base(new IntPtr(DefaultInvalidHandleValue), true) { }
```

Comment:
No clue if stuff like this gets inlined, but you could delegate to `this(IntPtr.Zero)` (or creating the new pointer).